python/ray/train/tests/test_gpu.py

```diff
@@ -111,14 +111,13 @@ def train_fn():
 
     count = Counter(devices)
     if num_gpus_per_worker == 0.5:
-        for i in range(4):
+        for i in range(2):
             assert count[i] == 4
     elif num_gpus_per_worker == 1:
-        for i in range(4):
+        for i in range(2):
             assert count[i] == 2
     elif num_gpus_per_worker == 2:
-        for i in range(2):
-            assert count[2 * i] == 2
+        assert count[0] == 2
```

Comment:
@amogkam  had a separate concern about the other test that you wrote -- https://github.com/ray-project/ray/pull/26493/files#r924064169

could you address that in this PR?

Reply:
> Hmm what exactly is this test supposed to be testing and how reliable is it? If we have 2 trials, each with 1 gpu (2*0.5) then the placement of the trials can be on either node, and we will get different results each time?

Example: in this case, if there is no preference of using the node for the ray. i agree with @amogkam , the results would be different. 
> node1 [0, 1], node2 [0, 1]

If the two trials are on the same node, it would give `[0, 0, 1, 1]`. 
If the two trials are on the different nodes, it would give `[0, 0, 0, 0]`. 

---------
The good thing about this test, `num_workers=int(4 / num_gpus_per_worker)` we always saturate the use of the gpus on the node. In this setup, the output might not be varied if there are launched at the same time.

Reply:
Right for this test we are fully saturating the GPUs, but I'm referring to the test in `test_tune`. 

For that test we've hardcoded `nums_samples=2` and `num_workers: 2`. 

Also, what is that test testing for? Is it needed if we have this test in `test_gpu` already?